hub/core/tests/test_storage_provider.py

```diff
@@ -34,18 +40,197 @@ def test_provider(provider=MappedProvider()):
     with pytest.raises(KeyError):
         provider[FILE_1]
 
+    provider.flush()
+
 
 def test_memory_provider():
-    test_provider(MemoryProvider("abc/def"))
+    check_storage_provider(memory_provider)
 
 
 def test_local_provider():
-    test_provider(LocalProvider("./hub_storage_local_test"))
+    check_storage_provider(local_provider)
 
 
 @pytest.mark.skipif(not s3_creds_exist(), reason="requires s3 credentials")
 def test_s3_provider():
-    test_provider(S3Provider("snark-test/hub_storage_s3_test"))
+    check_storage_provider(s3_provider)
+
+
+def test_lru_mem_local(benchmark):
+    lru = get_cache_chain([memory_provider, local_provider], [32 * MB])
+    check_storage_provider(lru)
```

Comment:
I don't think these assertions are enough to test the cache. We'll also need to make sure it's flushing at the right times, and handling its size restriction appropriately.

Reply:
added more detailed cache tests